static/js/tools/timeline/data_fetcher.ts

```diff
@@ -430,60 +319,88 @@ export function getStatData(
  */
 export function statDataFromModels(
   mainStatData: StatData,
-  modelStatAllResponse: TimelineStatAllData,
+  modelStatAllResponse: SeriesAllApiResponse,
   modelStatVars: string[]
 ): [StatData, StatData] {
-  const modelData = {
+  const modelData: StatData = {
     places: [],
     statVars: [],
     dates: [],
     data: {},
     sources: new Set<string>(),
     measurementMethods: new Set<string>(),
+    facets: {},
   };
   if (!mainStatData.dates.length) {
     return [mainStatData, modelData];
   }
-  for (const place in modelStatAllResponse) {
-    const placeData = modelStatAllResponse[place];
-    modelData.places.push(place);
-    modelData.data[place] = { data: {} };
-    for (const sv of modelStatVars) {
-      if (!(place in mainStatData.data)) {
+  const mainFacets = _.cloneDeep(mainStatData.facets);
+  mainStatData.facets = {};
+  const allFacets = modelStatAllResponse.facets;
+  for (let index = 0; index < modelStatVars.length; index++) {
+    const sv = modelStatVars[index];
+    const svData = modelStatAllResponse.data[sv];
+    modelData.statVars.push(sv);
+    const computedFacetId = index.toString();
+    for (const place in svData) {
+      if (!(place in mainStatData.data[sv])) {
         continue;
       }
-      const mainObsPeriod =
-        mainStatData.data[place].data[sv].metadata.observationPeriod;
+      modelData.places.push(place);
       modelData.statVars.push(sv);
```

Comment:
I think we want to remove this line because otherwise, we would be adding sv to modelData.statVars as many times as there are number of places in the data

Reply:
Yes.. this is a left over statement.